### PR TITLE
Add 1.16 jobs and specify specific K8s patch versions

### PR DESCRIPTION
**Pull request by msau42**

### CODE REVIEW

**Comment by msau42** [comment]
Testing 1.16.2 in https://github.com/kubernetes-csi/csi-driver-host-path/pull/100

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 1]
```diff
@@ -265,7 +273,7 @@ EOF
                         cat >>"$base/$repo/$repo-config.yaml" <<EOF
   - name: $(job_name "pull" "$repo" "$tests" "$deployment" "$kubernetes")
     always_run: $(pull_alwaysrun "$tests")
-    optional: $(pull_optional "$tests")
+    optional: $(if [ "$kubernetes" = "1.16.2" ]; then echo true; else pull_optional "$tests"; fi)
```
Comment (pohly) [inline config/jobs/kubernetes-csi/gen-jobs.sh]: This special case is not obvious and deserves a comment. Perhaps change this into `optional: $(pull_optional "$kubernetes" "$tests")` and then inside the extended implementation of `pull_optional` explain why 1.16.2 is an exception?

Comment (msau42) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: done

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 2]
```diff
@@ -224,7 +232,15 @@ expand_tests () {
...
-    if [ "$1" == "alpha" ]; then
+    local tests="$1"
+    local kubernetes="$2"
+
+    if [ "$tests" == "alpha" ]; then
+        echo "true"
+    elif [ "$kubernetes" == "1.16.2" ]; then
+        # Testing 1.16 requires release-tools to be updated in all
+        # kubernetes-csi repos. Once that is done, and tests
+        # are passing, this can be removed.
```
Comment (pohly) [inline config/jobs/kubernetes-csi/gen-jobs.sh]: We are almost done with that. Only https://github.com/kubernetes-csi/node-driver-registrar/pull/53 and https://github.com/kubernetes-csi/external-snapshotter/pull/188 are pending.

**Pull request merged**